src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -40,31 +41,57 @@ public AutoMLExperiment(MLContext context, AutoMLExperimentSettings settings)
             }
 
             _serviceCollection = new ServiceCollection();
+            InitializeServiceCollection();
         }
 
         private void InitializeServiceCollection()
         {
             _serviceCollection.TryAddTransient((provider) =>
             {
+                var channel = provider.GetRequiredService<IChannel>();
                 var context = new MLContext(_settings.Seed);
-                _cts.Token.Register(() =>
+                _globalCancellationTokenSource.Token.Register(() =>
```

Comment:
>_globalCancellationTokenSource [](http://example.com/codeflow?start=16&length=30)

would this instance get created at this point?

Reply:
Yep